core/runtime/src/main/java/io/quarkiverse/langchain4j/runtime/aiservice/AiServiceMethodImplementationSupport.java

```diff
@@ -642,6 +644,23 @@ private static UserMessage prepareUserMessage(AiServiceContext context, AiServic
         }
     }
 
+    private static Map<String, Object> getTemplateVariables(Object[] methodArgs,
+            AiServiceMethodCreateInfo.UserMessageInfo userMessageInfo) {
+        Map<String, Object> variables = new HashMap<>();
+
+        if (userMessageInfo.template().isPresent()) {
+            AiServiceMethodCreateInfo.TemplateInfo templateInfo = userMessageInfo.template().get();
+            Map<String, Integer> nameToParamPosition = templateInfo.nameToParamPosition();
+
+            for (var entry : nameToParamPosition.entrySet()) {
+                Object value = transformTemplateParamValue(methodArgs[entry.getValue()]);
+                variables.put(entry.getKey(), value);
+            }
+        }
+
+        return variables;
```

Comment:
Can it be `null` as mentioned in the javadoc? Looks like it's always there.

Reply:
Indeed, we will always have a map, but could be empty.